x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/privilege/IndexPrivilege.java

```diff
@@ -132,6 +138,21 @@ public final class IndexPrivilege extends Privilege {
     );
     private static final Automaton AUTO_CONFIGURE_AUTOMATON = patterns(AutoPutMappingAction.NAME, AutoCreateAction.NAME);
 
+    private static final Automaton CROSS_CLUSTER_REPLICATION_AUTOMATON = patterns(
+        "indices:data/read/xpack/ccr/shard_changes*",
+        IndicesStatsAction.NAME + "*",
+        RetentionLeaseActions.Add.ACTION_NAME + "*",
+        RetentionLeaseActions.Remove.ACTION_NAME + "*",
+        RetentionLeaseActions.Renew.ACTION_NAME + "*"
+    );
```

Comment:
With this new tightly scope privilege, CCR is no longer a super set privilege of CCS. I think this is actually a good thing. If both ccr and ccs are needed for the same index, we just need to grant both privileges.

Previously, I also talked about creating a new CCS privilege which basically covers both `read` and `read_cross_cluster`. But I decided against it in this PR because (1) it does not add a lot value (1 new named privileges vs 2 existing ones) and (2) this new privilege, say `cross_cluster_search` would be suggested in authorization denial messages when any `read` action is denied. I think that feels quite distracting. What do you think?